src/ensembl/src/content/app/browser/Browser.tsx

```diff
@@ -118,16 +122,13 @@ export const Browser = (props: BrowserProps) => {
   }, [props.activeGenomeId, props.activeEnsObjectId]);
 
   useEffect(() => {
-    const {
-      browserQueryParams: { location }
-    } = props;
     const { genomeId } = params;
     const chrLocation = location ? getChrLocationFromStr(location) : null;
 
     if (props.browserActivated && genomeId && chrLocation) {
       props.changeBrowserLocation({ genomeId, chrLocation, ensObjectId: null });
     }
-  }, [props.browserActivated]);
+  }, [props.browserActivated, location]);
```

Comment:
Wondering whether the logic of this useEffect callback belongs in useBrowserRouting.

Reply:
I have moved it to `useBrowserRouting`. Please check if it looks right.